test/behavior/ptrcast.zig

```diff
@@ -80,9 +79,34 @@ fn testReinterpretBytesAsExternStruct() !void {
     try expect(val == 5);
 }
 
-test "reinterpret bytes of an extern struct into another" {
+test "reinterpret bytes of an array into an over-sized extern struct" {
+    if (builtin.zig_backend == .stage1) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_x86_64) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_arm) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_aarch64) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_wasm) return error.SkipZigTest; // TODO
+    if (builtin.zig_backend == .stage2_c) return error.SkipZigTest; // TODO: Fix UB below
+
+    const bytes align(4) = [_]u8{ 1, 2, 3, 4 };
+    const S = struct {
+        a: u16,
+        b: u16,
+        c: u32,
+    };
+
+    const x = @ptrCast(*const S, &bytes);
+    var p = &x.b; // TODO: Triggers ubsan for CBE, since @sizeOf(S) > @sizeOf(bytes)
+
+    const expected = switch (native_endian) {
+        .Little => 0x0403,
+        .Big => 0x0304,
+    };
+    try expect(p.* == expected);
```

Comment:
This is not a valid Zig code. The struct does not have a well-defined layout. What's the goal of adding this test case?

Reply:
Should have been an extern struct, sorry about that!

Reply:
With an extern struct, it's at best a grey area of the spec whether this will be allowed or not, for the same reason as it is not allowed in C.

Reply:
Would you prefer we leave it out for now, or a TODO to decide later whether this is spec'd or not?